Magicx XU mini Support - Initial draft

olivierschonken opened the pull request:
Putting this here as a placeholder. Devicetree can be refactored to piggyback on XU10 devicetree more. Still need to add boot.ini devicetree selection (currently modifying boot.ini to load the magicx-xu-mini devicetree)

Have this working on top of the [20240815](https://github.com/ROCKNIX/distribution/releases/tag/20240815) release, but having issues on startup when rebased on current dev branch.

olivierschonken commented:
Done :+1:

olivierschonken commented:
> Can you dump the ADC values? That will help us define the device in the boot.ini:
> 
> `for RAW in $(find /sys/devices/platform/ -name "in_*_raw"); do echo ${RAW}; cat ${RAW}; done`

555
519
531
517
1022
518

r3claimer commented:
just post everything.... the entire output

olivierschonken commented:
> just post everything.... the entire output

Updated the values :+1:

sydarn commented:
That looks the same as xu10: https://github.com/ROCKNIX/hardkernel-uboot/blob/RK3326/cmd/hwrev.c#L48

Is there any gpio trick you can do like on rgb10 here: https://github.com/ROCKNIX/distribution/blob/8c7be175ee26ed5be35ca1b215b3137e1136525d/projects/Rockchip/devices/RK3326/bootloader/boot.ini#L13

olivierschonken commented:
> That looks the same as xu10: https://github.com/ROCKNIX/hardkernel-uboot/blob/RK3326/cmd/hwrev.c#L48
> 
> Is there any gpio trick you can do like on rgb10 here:
> 
> https://github.com/ROCKNIX/distribution/blob/8c7be175ee26ed5be35ca1b215b3137e1136525d/projects/Rockchip/devices/RK3326/bootloader/boot.ini#L13

I tried finding a workaround in terms of pinstate at bootup, but it didn't seem to work for differentiating via gpio input. Also don't have a XU10 to verify if I had something that worked for IDing the xu mini.  For now I added it as a commented out load under the xu10 hwrev. So people will have to comment it out when using ROCKNIX with the device. Looks like this was also done for rk3326-powkiddy-rgb20s

olivierschonken commented:
Updated the PR to re-use as much as possible of the xu10 devicetree, also reworked the key bindings.  With Retroarch the cores work better when SDL is set as the audio provider, Is there a way to specify that in a quirk?

sydarn commented:
Sorry for the delay on this, I think we have decided to make a 2nd image with rk3326 devices. Will just take some time, but that would mean the XU mini can be uniquely identified in that image.

olivierschonken commented:
> Sorry for the delay on this, I think we have decided to make a 2nd image with rk3326 devices. Will just take some time, but that would mean the XU mini can be uniquely identified in that image.

All good.  I've been keeping the branch updated with dev, so hopefully it should easily slot into that image then.

olivierschonken commented:
Made the changes, but somehow on my local build something is not starting up proper with the b image.  Will try and debug, maybe my build just got out of sync somewhere

olivierschonken commented:
Tried the image without success.  If I flash the a image and copy b_boot.ini to ROCKNIX/boot.in and rk3326-magicx-xu-mini.dtb into ROCKNIX, it succesfully boots up.  So something goes missing when building the b image, and I can't quite put my finger on it.  Maybe something u-boot related?

olivierschonken commented:
Unfortunately I currently have no serial or ssh access.  Dev was mostly done with flashing images on the SD card, and writing logs to the sd card when possible.  For the B Image I can see it doesn't start up at all - i.e. backlight and LCD and LED remains in state similar to when no SD card inserted.  Only way I could boot it up was as mentioned to Flash the A image, and put the correct devicetree and b_boot.ini on the card.

olivierschonken commented:
MagicX XU Mini M, image attached

On Thu, 26 Sept 2024, 08:42 spycat88, ***@***.***> wrote:

> Can you go into system information in ES and see what it shows the device
> as ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ROCKNIX/distribution/pull/511#issuecomment-2376064503>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIX6KQS53Q36DXALIRRFJLZYOUFLAVCNFSM6AAAAABNQ73GHSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGNZWGA3DINJQGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

olivierschonken commented:
![Uploading 20240926_092455.jpg…]()

olivierschonken commented:
![20240926_092455](https://github.com/user-attachments/assets/d52c4f03-8788-4737-a8d9-9b16a87c1f9e)

olivierschonken commented:
Adding the rk3326-magicx-xu10 on b section helps to get it to boot, had to revert 28af680bb35db26da114f960f9578ef946bed2cc on current dev to get the display to work though...
